flags/flags.go

```diff
@@ -0,0 +1,77 @@
+package flags
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/launchdarkly/ld-find-code-refs/aliases"
+	"github.com/launchdarkly/ld-find-code-refs/internal/helpers"
+	"github.com/launchdarkly/ld-find-code-refs/internal/ld"
+	"github.com/launchdarkly/ld-find-code-refs/internal/log"
+	"github.com/launchdarkly/ld-find-code-refs/internal/version"
+	"github.com/launchdarkly/ld-find-code-refs/options"
+)
+
+const (
+	minFlagKeyLen = 3 // Minimum flag key length helps reduce the number of false positives
+)
+
+func GenerateSearchElements(opts options.Options, repoParams ld.RepoParams) ([]string, map[string][]string) {
+
+	projKey := opts.ProjKey
+
+	ldApi := ld.InitApiClient(ld.ApiOptions{ApiKey: opts.AccessToken, BaseUri: opts.BaseUri, ProjKey: projKey, UserAgent: "LDFindCodeRefs/" + version.Version})
+	isDryRun := opts.DryRun
+
+	ignoreServiceErrors := opts.IgnoreServiceErrors
+	if !isDryRun {
+		err := ldApi.MaybeUpsertCodeReferenceRepository(repoParams)
+		if err != nil {
+			helpers.FatalServiceError(err, ignoreServiceErrors)
+		}
+	}
+
+	flags, err := getFlags(ldApi)
+	if err != nil {
+		helpers.FatalServiceError(fmt.Errorf("could not retrieve flag keys from LaunchDarkly: %w", err), ignoreServiceErrors)
+	}
+
+	filteredFlags, omittedFlags := filterShortFlagKeys(flags)
+	if len(filteredFlags) == 0 {
+		log.Info.Printf("no flag keys longer than the minimum flag key length (%v) were found for project: %s, exiting early",
+			minFlagKeyLen, projKey)
+		os.Exit(0)
+	} else if len(omittedFlags) > 0 {
+		log.Warning.Printf("omitting %d flags with keys less than minimum (%d)", len(omittedFlags), minFlagKeyLen)
+	}
+
+	aliases, err := aliases.GenerateAliases(filteredFlags, opts.Aliases, opts.Dir)
+	if err != nil {
+		log.Error.Fatalf("failed to create flag key aliases: %v", err)
+	}
+
+	return filteredFlags, aliases
+}
+
+// Very short flag keys lead to many false positives when searching in code,
+// so we filter them out.
+func filterShortFlagKeys(flags []string) (filtered []string, omitted []string) {
+	filteredFlags := []string{}
+	omittedFlags := []string{}
```

Comment:
Nit: goland gave me warnings for these saying: `Empty slice declaration using a literal `. It made these suggestions which seem to make sense:

```go
   var filteredFlags []string
   var omittedFlags []string
```

Reply:
I just tried changing to what you suggested and also naming the return `filteredFlags` and `omittedFlags` both have caused tests to fail.